src/node.rs

```diff
@@ -577,35 +569,62 @@ impl EventStepper for Node {
 
 #[cfg(feature = "mock_base")]
 impl Node {
-    /// Returns the list of banned clients' IPs held by this node.
-    pub fn get_banned_client_ips(&self) -> BTreeSet<IpAddr> {
-        self.machine.current().get_banned_client_ips()
+    /// Returns the chain for this node.
+    pub fn chain(&self) -> Option<&Chain> {
+        self.machine.current().chain()
+    }
+
+    /// Returns this node state.
+    pub fn node_state(&self) -> Option<&crate::states::Node> {
```

Comment:
I'm going to put this comment here, but it involves other changes in this PR as well:
I'm not sure I see the benefit of exposing the internal state like this and removing the forwarding functions. It seems to go contrary to what's generally considered good practices: separation of concerns and low coupling.

Reply:
I agree with this in general, but I do think testing have its own requirements that do not always fit well with all the other code good practices.

I'm not sure the low coupling benefit is really there, when we are exposing internals through multiple layer of indirection.
Right now this is more making it extra convoluted to things when needed, and resulted in 3 different way of accessing that internals.

Reply:
I'd prefer to have only one way to do things and that way to be the forwarding methods to expose only what's really necessary. That said, I don't consider this to be that important in the end and with your approach at least there is less code which is always nice. I'm OK with it.